iceoryx_posh/test/fuzztests/cmdlineparserfuzzing.cpp

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "cmdlineparserfuzzing.hpp"
+#include "fuzz_helper.hpp"
+#include "iceoryx_posh/internal/log/posh_logging.hpp"
+#include "iceoryx_utils/platform/getopt.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+
+iox::log::LogLevel m_logLevel{iox::log::LogLevel::kOff};
+CmdLineParserFuzzing::CmdLineParserFuzzing()
```

Comment:
@elfenpiff Thank you very much for the review. I agree with you that noexcept makes sense for Iceoryx in general. However, the fuzz wrappers should only be used for testing purposes. Within the test, it could also make sense to throw an exception. I also used the heap for the fuzz wrappers, knowing that this is also not allowed in general and I 100% agree on that. For my understanding though, the fuzz wrappers shall be used as an help to perform additional tests against Iceoryx and are therefore not really part of Iceoryx itself. However, if you think noexcept is still necessary in this case, I could try to add it with no doubt.

Reply:
@JMueller my thought here was that nowhere in iceoryx are exceptions handled, even not in unit tests and I doubt we will start with this since it just adds complexity. The noexcept keyword on every method causes the application to die instantly when an exception is thrown inside this method and that the exception is not propagated upwards. It does not prevent exceptions per se.

It makes it also clear for the future you and me that we should never start some kind of exception handling in a unit test or some other construct - either we do it right away inside the method or we state that we die early. And in most cases, it is what we want for instance if we encounter a `std::bad_alloc` exception when we are out of memory.

So you are completely right with your assessment. But either we handle exceptions right away or never which both leads to a noexcept keyword at the method/function.

Reply:
Done.